reana_job_controller/app.py

```diff
@@ -55,212 +59,195 @@ def filter_jobs(job_db):
         job['obj'] = job_obj
         job['deleted'] = False
-        JOB_DB[job_id] = job
-        return jsonify({'job-id': job_id}), 201
+        JOB_DB[job['job_id']] = job
+        return jsonify({'job_id': job['job_id']}), 201
     else:
         return jsonify({'job': 'Could not be allocated'}), 500
 
 
 @app.route('/jobs/<job_id>', methods=['GET'])
-def get_job(job_id):
-    """Get a job.
-
-    FIXME --> probably this endpoint should be merged with `get_jobs()`
-
-    .. http:get:: /jobs/<job_id>
-
-        Returns a JSON list with all the jobs.
-
-        **Request**:
-
-        .. sourcecode:: http
-
-            GET /jobs/cdcf48b1-c2f3-4693-8230-b066e088c6ac HTTP/1.1
-            Content-Type: application/json
-            Host: localhost:5000
-
-        :reqheader Content-Type: application/json
-
-        **Responses**:
-
-        .. sourcecode:: http
-
-            HTTP/1.0 200 OK
-            Content-Length: 80
-            Content-Type: application/json
-
-            {
+def get_job(job_id):  # noqa
+    """Get a Job.
+
+    ---
+    get:
+      description: Get a Job by its id
+      produces:
+       - application/json
+      parameters:
+       - name: job_id
+         in: path
+         description: ID of the Job
+         required: true
+         type: string
+      responses:
+        200:
+          description: The Job.
+          schema:
+            $ref: '#/definitions/Job'
+          examples:
+            application/json:
               "job": {
                 "cmd": "sleep 1000",
                 "cvmfs_mounts": [
                   "atlas-condb",
                   "atlas"
                 ],
-                "docker-img": "busybox",
+                "docker_img": "busybox",
                 "experiment": "atlas",
-                "job-id": "cdcf48b1-c2f3-4693-8230-b066e088c6ac",
+                "job_id": "cdcf48b1-c2f3-4693-8230-b066e088c6ac",
                 "max_restart_count": 3,
                 "restart_count": 0,
                 "status": "started"
               }
-            }
-
-        :resheader Content-Type: application/json
-        :statuscode 200: no error - the list has been returned.
-        :statuscode 404: error - the specified job doesn't exist.
+        404:
+          description: The Job does not exist.
     """
+
     if job_id in JOB_DB:
         job_copy = copy.deepcopy(JOB_DB[job_id])
         del(job_copy['obj'])
         del(job_copy['deleted'])
         if job_copy.get('pod'):
             del(job_copy['pod'])
+
+        # FIXME job_schema.dump(job_copy) no time now to test
+        # since it needs to be run inside the cluster
         return jsonify({'job': job_copy}), 200
     else:
-        abort(404)
+        return jsonify({'message': 'The job {} doesn\'t exist'.
+                                   format(job_id)}), 400
 
 
+@app.route('/apispec', methods=['GET'])
+def get_openapi_spec():
+    """Get OpenAPI Spec.
+
+    FIXME add openapi spec
```

Comment:
Absolutely, there are things such as validating OpenAPI spec file which is generated in `run_tests.sh` which is far from being acceptable. I will open issues 👍